homeassistant/components/evohome.py

```diff
@@ -8,138 +8,153 @@
 """
 
 # Glossary:
-# TCS - temperature control system (a.k.a. Controller, Parent), which can
-# have up to 13 Children:
-#   0-12 Heating zones (a.k.a. Zone), and
-#   0-1 DHW controller, (a.k.a. Boiler)
+#   TCS - temperature control system (a.k.a. Controller, Parent), which can
+#   have up to 13 Children:
+#     0-12 Heating zones (a.k.a. Zone), and
+#     0-1 DHW controller, (a.k.a. Boiler)
+# The TCS & Zones are implemented as Climate devices, Boiler as a WaterHeater
 
 import logging
 
 from requests.exceptions import HTTPError
 import voluptuous as vol
 
 from homeassistant.const import (
-    CONF_USERNAME,
-    CONF_PASSWORD,
-    CONF_SCAN_INTERVAL,
-    HTTP_BAD_REQUEST
+    CONF_SCAN_INTERVAL, CONF_USERNAME, CONF_PASSWORD,
+    EVENT_HOMEASSISTANT_START,
+    HTTP_BAD_REQUEST, HTTP_SERVICE_UNAVAILABLE, HTTP_TOO_MANY_REQUESTS
 )
-
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
 
-REQUIREMENTS = ['evohomeclient==0.2.7']
-# If ever > 0.2.7, re-check the work-around wrapper is still required when
-# instantiating the client, below.
+REQUIREMENTS = ['evohomeclient==0.2.8']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'evohome'
 DATA_EVOHOME = 'data_' + DOMAIN
+DISPATCHER_EVOHOME = 'dispatcher_' + DOMAIN
 
 CONF_LOCATION_IDX = 'location_idx'
-MAX_TEMP = 28
-MIN_TEMP = 5
-SCAN_INTERVAL_DEFAULT = 180
-SCAN_INTERVAL_MAX = 300
+SCAN_INTERVAL_DEFAULT = 300
+SCAN_INTERVAL_MINIMUM = 180
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_LOCATION_IDX, default=0): cv.positive_int,
+        vol.Optional(CONF_LOCATION_IDX, default=0):
+            cv.positive_int,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL_DEFAULT):
+            vol.All(vol.Coerce(int), vol.Range(min=SCAN_INTERVAL_MINIMUM)),
```

Comment:
Scan interval should be `cv.time_period`.

Reply:
If I do it this way, then voluptuous does the range check (`min=SCAN_INTERVAL_MINIMUM`), else I have to move it into the `setup()`.

Or can I add a minimum value to `cv.time_period`?
```python
        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL_DEFAULT):
            vol.All(cv.time_period, vol.Range(min=SCAN_INTERVAL_MINIMUM)),
```